packages/fxa-content-server/app/scripts/models/auth_brokers/mixins/channel.js

```diff
@@ -73,7 +77,7 @@ const ChannelMixin = {
     const channel = this.getChannel();
     const send = ensureActionReturnsPromise(channel.send.bind(channel));
 
-    return send(message, data);
+    return this._metrics.flush().then(() => send(message, data));
```

Comment:
nit: We could update comment block to mention that metrics are flushed upon sending.

Reply:
Good suggestion, will do.